### PR TITLE
fix(doctor): report what SERVES, not the stale legacy credential (v5.5.63)

### PR DESCRIPTION
**Pull request by askalf**
`dario doctor` printed **`OAuth expired` while the proxy was answering 200s** on both Haiku and Sonnet — and `dario-doctor-watch` filed an issue for it on every run. #1105 was opened during exactly that healthy state.

## Cause
The OAuth row read **only** the legacy `credentials.json`. But dario#805 deliberately **keeps a newer pool token and refuses to overwrite that file**, so "legacy is stale" is an *expected steady state* — every recovery that restores a pool account leaves one behind. Meanwhile the separate **Pool** row was correctly reporting `1/1 healthy` two lines below. The doctor contradicted itself, and the watcher believed the wrong line.

## Fix
The row consults the **pool** first:
- live pool → `ok`, and it still **names the stale legacy file and cites #805** — disclosing rather than hiding, so the next reader knows why that file is left alone;
- nothing able to serve → unchanged `warn`/`fail`. Silencing a real outage would be worse than the false alarm it replaces.

Extracted as the pure **`oauthCheckRow()`**, mirroring the existing `checkIdentityDrift()` so the branch logic is unit-testable without touching the filesystem — 13 new assertions covering the #1105 case, pluralisation, both failure modes, and "an empty pool is not a live pool". **144/144.**

Verified against the box's real state (pool live + legacy expired): previously `WARN OAuth expired`, now `ok — pool credential live (1/1 account), the legacy credentials.json is expired and unused (dario#805)`.

### CODE REVIEW

**Review by sprayberry-redline — CHANGES_REQUESTED** [review]
Automated review from the Sprayberry Labs fleet code reviewer.

**Verdict:** 1 blocking issue (a self-inflicted CHANGELOG corruption) — the `doctor.ts` logic itself is solid and well-tested; requesting changes just for the changelog file.

## Findings

### 1. [Blocking] CHANGELOG.md: new release entry lands inside the HTML template comment, never renders

`CHANGELOG.md` diff:
```
-time, rename that heading to `## [X.Y.Z] - YYYY-MM-DD` and add a fresh
+time, rename that heading to `## [5.5.63] - 2026-08-25
+
+- **Fixed: `dario doctor` reported `OAuth expired` while the proxy was serving** (#1105). ...
+
+## [X.Y.Z] - YYYY-MM-DD` and add a fresh
 `## [Unreleased]` above it. See CONTRIBUTING for the full release
 checklist.
 -->
```
The whole 5.5.63 entry (and the "rename the heading" instruction sentence around it) got inserted between lines 5 (`<!--`) and 14 (`-->`) — i.e. *inside* the HTML comment block, not after it. Concretely: the file still opens the comment at line 5, the new entry sits between the truncated sentence "...rename that heading to `## [5.5.63] - 2026-08-25" and the re-appended "`## [X.Y.Z] - YYYY-MM-DD` and add a fresh", and the comment still closes at `-->` a few lines later. Rendered on GitHub, the whole entry is invisible (HTML comments don't render), `## [Unreleased]` right below is left empty, and there is no actual `## [5.5.63] - 2026-08-25` heading anywhere in the file. It also corrupts the template instructions themselves for the next contributor, since the placeholder `## [X.Y.Z] - YYYY-MM-DD` line now reads as a leftover fragment rather than the reusable example.

```suggestion
## [Unreleased]

## [5.5.63] - 2026-08-25

- **Fixed: `dario doctor` reported `OAuth expired` while the proxy was serving** (#1105). ...
```
placed *after* the `-->` closing the template comment, restoring the original comment body unchanged.

## What's good
- `oauthCheckRow()` in `src/doctor.ts:155-192` is a clean, pure extraction of the branch logic mirroring `checkIdentityDrift()`, with the pool-first / legacy-fallback precedence clearly justified and the "nothing can serve" path still surfaced as warn/fail rather than silenced.
- `test/doctor-oauth-pool.mjs` covers the #1105 regression case, pluralization, both failure modes, and the "empty pool isn't a live pool" edge case — good coverage for a bugfix PR.
- CI (`triage`) is green; no other checks are blocking.

Package version/lockfile bump (5.5.61 → 5.5.63) and `test:serial` script addition of the new test file are consistent with the change.

**Review by sprayberry-redline — DISMISSED** [review]
Automated review from the Sprayberry Labs fleet code reviewer. Re-review of head `3b1b92b` — a follow-up to my earlier CHANGES_REQUESTED review.

**Verdict:** the prior blocking issue is fixed; no new blocking findings — approving.

## Follow-up on the prior finding

The CHANGELOG entry no longer lands inside the HTML template comment. Verified in the current head: `## [Unreleased]` is immediately followed by a real `## [5.5.64] - 2026-08-25` heading, both outside and after the `<!-- ... -->` release-convention block, so the entry now renders correctly on GitHub.

## What I checked this pass

- `src/doctor.ts:155-192` (`oauthCheckRow`) and its call site at `src/doctor.ts:628-645`: the pool-first / legacy-fallback precedence matches the 13 assertions in `test/doctor-oauth-pool.mjs`, including the "empty pool isn't a live pool" edge case (`poolHealthy > 0` gate, not `poolTotal > 0`).
- The `try { ... } catch { /* pool unreadable — oauthCheckRow falls back to the legacy verdict */ } ` wrapper around `accounts.js` import correctly leaves `poolHealthy`/`poolTotal` at `0` on failure, which routes to the legacy warn/fail branch as intended — no swallowed failure that would misreport `ok`.
- CI: `compat` shows FAILURE, but `gh run list` for the same workflow shows unrelated concurrent commits (`chore(cc-drift): v5.5.62`, a dedicated `fix(ci): compat must not touch the shared Claude credential` PR) failing with the identical `invalid_grant` / "Refresh token not found or invalid" error in the same time window — this is a known, pre-existing shared-credential CI issue, not something this diff introduces. All other checks (test, build, CodeQL, actionlint, triage, validate-package-json, docker-cap-drop-smoke) are green.

## What's good

- Clean re-fix: the changelog surgery was done correctly this time, entry content unchanged.
- `oauthCheckRow` stays a pure, well-tested function; the integration point is minimal and doesn't change behavior on the `s.authenticated === true` path.

No blocking issues found on this pass.

**Comment by sprayberry-redline** [comment]
## dario/live-test — PASS

Head SHA: `3b1b92b2835e0f3b7eaab700e703bb77eaf6a88a`

| stage | result | note |
|---|---|---|
| install (`npm ci`) | ✅ pass | clean, 6 packages added |
| build (`npm run build`) | ✅ pass | tsc clean |
| test (`npm test`) | ✅ pass | 144/144 passed, 0 fail |
| cli (`--help`) | ✅ pass | exit 0 |
| doctor | ✅ pass | reports `[FAIL] OAuth not authenticated` as expected with no credentials in the scrubbed env — correct report, not a crash (exit 1 is doctor's normal "issues found" exit) |
| live probe | ⏭️ skipped | `DARIO_TEST_UPSTREAM_API_KEY` not set in this environment |

**Review by sprayberry-redline — DISMISSED** [review]
**Automated review from the Sprayberry Labs fleet code reviewer.**

**Verdict:** No blocking issues found — clean, well-tested fix. Approving.

## What I checked
- `gh pr diff 1107` in full (`src/doctor.ts`, new `test/doctor-oauth-pool.mjs`, `package.json`/`package-lock.json` version bump, `CHANGELOG.md`).
- CI check-rollup (`gh pr view --json statusCheckRollup`) and the failing `compat` job's log.

## Correctness
The new `oauthCheckRow()` (`src/doctor.ts`) is a pure function with a clear precedence: a live pool account (`poolHealthy > 0`) reports `ok` while still disclosing the stale legacy file and citing dario#805; otherwise it falls back to the legacy `legacyStatus`/`legacyCanRefresh` verdict unchanged. The call site in `runChecks()` wraps the pool lookup in `try/catch`, defaulting `poolHealthy`/`poolTotal` to `0` on any read failure, which safely degrades to the pre-existing legacy-only behavior rather than throwing or claiming false health. The new pluralization and "empty pool ≠ live pool" cases are covered by the 13 added assertions in `test/doctor-oauth-pool.mjs`, and the CI `test` job is green.

## CI check-rollup
One check, `compat` (self-hosted), is FAILURE. The log shows the proxy failed to reach `/health` within 30s because of a runner-level OAuth refresh error: `Startup refresh failed for login: Refresh failed for login (400): invalid_grant`. This is a runner/credential-state issue, not something introduced by this diff (`doctor.ts`/`doctor-oauth-pool.mjs` don't touch proxy startup or refresh). The same `compat` check has failed independently on other recent PRs (#1101, #1099) with green `test`/`build`, consistent with a pre-existing flaky self-hosted runner rather than a regression from this change. Not treating it as blocking, but flagging it here for visibility.

## What's good
- The `Check` object retains the legacy file's status in its `detail` string even in the healthy-pool branch, rather than hiding it — avoids silently masking a real outage from #805.
- Function extraction mirrors the existing `checkIdentityDrift()` pattern, keeping the branch logic unit-testable without filesystem access.

**Comment by sprayberry-redline** [comment]
## dario/live-test — PASS

Head SHA: `9154505c4fe4b8eaa7063de29bfecf24d716bdbf`

| stage | result | note |
|---|---|---|
| install | PASS | `npm ci` clean |
| build | PASS | `tsc` clean |
| test | PASS | 144/144 tests passed |
| cli | PASS | `--help` OK |
| doctor | PASS | no crash; correctly reports `[FAIL] OAuth not authenticated` in credential-less sandbox (expected, not a bug) |
| live probe | SKIPPED | no `DARIO_TEST_UPSTREAM_API_KEY` configured |

**Comment by askalf** [comment]
Ticket `01M0WSYADKMQ7TTCT00XYQ1A8V` was auto-filed from the 14:44:58Z CHANGES_REQUESTED review (CHANGELOG entry inside the leading HTML comment). **Verified at head `9154505`: already fixed — no change needed, closing the ticket without a patch.**

The entry now sits at line 14, below the `-->` (line 10) and below `## [Unreleased]` (line 12), so `## [5.5.64]` is a real rendering heading. `git diff origin/master...HEAD -- CHANGELOG.md` puts the added hunk at `@@ -11,6 +11,9 @@` — outside the comment block. Commit `3b1b92b` ("Merge master; re-bump to 5.5.64 and fix the changelog placement") landed the correction at 14:47:29Z, ~2.5 min after the review; the review was dismissed at 14:58:54Z and re-approved at 15:13:57Z. The ticket was cut at 15:52:07Z from the stale finding, 38 min after the approval. Version is 5.5.64 rather than the ticket's 5.5.63 because #1101 took 5.5.63 to master first.

One residual, flagged rather than actioned since it is outside this ticket: the PR is APPROVED and MERGEABLE but `mergeStateStatus=UNSTABLE`. **Compat test (self-hosted)** is red at this sha — the failing step is `Start dario proxy (passthrough mode)`, and `Run compat tests` never ran, so `COMPAT_EXIT_CODE` was empty and `Set job status` fell through to its `:-1` fail-closed default. That is the proxy-credential class #1106 addresses, not anything in this PR's diff. CI, CodeQL, and actionlint are green.

**Review by sprayberry-redline — CHANGES_REQUESTED** [review]
**Automated review from the Sprayberry Labs fleet code reviewer.**

**Verdict:** 1 blocking correctness issue (broken lockfile) — the `doctor.ts`/`oauthCheckRow` logic itself looks solid and well-tested.

## Findings

### 🔴 Blocking — `package-lock.json` contains unresolved merge-conflict markers (invalid JSON)

`package-lock.json` (diff hunk, lines ~1-12):
```
-  "version": "5.5.65",
+<<<<<<< HEAD
+  "version": "5.5.66",
+=======
+  "version": "5.5.66",
+>>>>>>> origin/master
```
and again a few lines below for the nested `packages[""].version` field. I confirmed this is committed as-is at the PR head (`ba89ca2`) via `gh api .../contents/package-lock.json` — the literal `<<<<<<< HEAD` / `=======` / `>>>>>>> origin/master` markers are in the file, not a diff-rendering artifact.

**Failure scenario:** `package-lock.json` is no longer valid JSON. Anyone running `npm ci` (the standard CI/deploy install command, which parses the lockfile strictly) against this commit gets a JSON parse error and a broken install/build. `npm install` may silently "fix" it by rewriting the file, masking the mistake locally but leaving the broken artifact in git history until someone regenerates it.

Note the PR's own CI shows `build`/`test` jobs green — worth double-checking whether CI actually runs `npm ci` against this lockfile or uses a cached `node_modules`, since a genuinely invalid lockfile would normally fail that step outright.

```suggestion
Resolve the conflict properly: keep a single `"version": "5.5.66"` in both spots (no markers), e.g. regenerate with `npm install` after merging master cleanly, then commit the resulting valid package-lock.json.
```

## What's good
- `oauthCheckRow()` is a clean, pure extraction that mirrors the existing `checkIdentityDrift()` pattern — easy to unit test without touching the filesystem.
- The pool-unreadable path is wrapped in `try/catch` and falls back to the legacy verdict rather than throwing, so a `listAccountAliases`/`loadAllAccounts` failure degrades gracefully instead of crashing `doctor`.
- New test file (`test/doctor-oauth-pool.mjs`) covers the live-pool-overrides-stale-legacy case, pluralization, and both "nothing can serve" failure modes — good coverage of the described bug.
- CI check-rollup is otherwise all green (CodeQL, compat, actionlint, multi-version build/test).

**Comment by sprayberry-redline** [comment]
## dario/live-test report — PR #1107 @ ba89ca29c0a457f2020df39421129220e1b8b9a5

| stage | result | note |
|---|---|---|
| install (`npm ci`) | PASS | clean |
| build (`npm run build`) | PASS | tsc + asset copy, no errors |
| test (`npm test`) | PASS | 144/144 tests passing, 0 fail, 0 skipped |
| cli (`--help`) | PASS | exits 0 |
| doctor | PASS | exits 1 as expected with no creds — reports `[FAIL] OAuth not authenticated`, no crash |
| live probe | SKIPPED | `DARIO_TEST_UPSTREAM_API_KEY` not configured in this environment |

**Verdict: PASS**

Head SHA: `ba89ca29c0a457f2020df39421129220e1b8b9a5`
Status: dario/live-test = success

**Review by sprayberry-redline — APPROVED** [review]
Automated review from the Sprayberry Labs fleet code reviewer.

**Verdict:** Re-review at head `533205f` — the prior blocking finding is fixed; approving.

## Re-review context
This PR previously received a `CHANGES_REQUESTED` review (2026-08-25T14:44:58Z) flagging unresolved
merge-conflict markers in `package-lock.json` that made the file invalid JSON. Commit `533205f`
("fix: repair package-lock.json conflict markers (invalid JSON)") addresses this directly.

## Verification
- `package-lock.json` at `533205f`: no `<<<<<<<`/`=======`/`>>>>>>>` markers remain; `JSON.parse()`
  succeeds cleanly.
- `CHANGELOG.md` at `533205f`: the `5.5.66` entry sits correctly under `## [5.5.66] - 2026-08-25`,
  outside and after the `<!-- ... -->` template comment block (lines 5-9) — it renders as intended.
- CI check-rollup is all green: CodeQL, actionlint, analyze, build (18/20/22), compat,
  docker-cap-drop-smoke, test, triage, validate-package-json all `pass`.
- `src/doctor.ts` / `test/doctor-oauth-pool.mjs` logic (the actual `oauthCheckRow()` fix) was already
  sound in the earlier review and is unchanged by this fix-up commit.

## What's good
- The fix commit is minimal and targeted — only touches the corrupted lockfile lines, no scope creep.
- Author's own commit message documents root cause (sed-only version bump missed conflict markers)
  and verification method (JSON.parse + from-scratch `npm ci`), which matches what I confirmed independently.

No blocking issues remain.

**Comment by sprayberry-redline** [comment]
## dario/live-test report — PR #1107 @ `533205f`

| Stage | Result | Note |
|---|---|---|
| install (`npm ci`) | ✅ pass | clean, only benign `allow-scripts` warning for esbuild postinstall |
| build (`npm run build`) | ✅ pass | `tsc` + asset copy, no errors |
| test (`npm test`) | ✅ pass | 144/144 tests passed, 0 failed |
| cli (`--help`) | ✅ pass | exits 0, help text renders |
| doctor | ✅ pass | exits 1 as expected with no credentials (`[FAIL] OAuth not authenticated`) — reports cleanly, does not crash |
| live probe | ⏭️ skipped | `DARIO_TEST_UPSTREAM_API_KEY` not set in tester environment |

**Verdict: PASS**

Head SHA tested: `533205f97f20cee307024b5fd7e31ba43d09a1fe`
